LFS-856: Simplify CARDS runmodes

acrowthe opened the pull request:
- Rename run modes `cardiac` to `kids` ~~and `cardiac-rehab` to `cards4care`~~
  - Rename the pom artifactIds to match
  - Kept `cardiac-rehab` untouched for compatibility
  
Current runmodes:
- no runmodes specified: generic cards, with no forms
- `lfs`: Cards4LFS
- `kids`: Cards4Kids
- `cardiac_rehab`: Cards4Care

`nolfs` kept as a valid (now default) runmode so `nolfs,cardiac_rehab` is still valid for Cards4Care compatibility

acrowthe committed:
LFS-856: Simplify CARDS runmodes
- Move resource runmodes (`lfs`, `cardiac`, `cardiac_rehab`) to a single option
  - `noresources` is now the default, which does not include any questionnaires
- Rename run modes `cardiac` to `cards4kids` and `cardiac-rehab` to `cards4care`
  - Rename the pom artifactIds to match
  - Rename the cardiac rehab service user to match

marta- reviewed (approved):
Looks good and is much easier to use.

I noticed that the artifact names changed, but the source directories were not renamed accordingly.

sashaandjic reviewed (approved):
tested in all 4 modes - all good

acrowthe commented:
@marta- 
> I noticed that the artifact names changed, but the source directories were not renamed accordingly.

I didn't want to rename every single file in the resource directories and potentially cause merge issues, but the artifact names were mentioned in the provisioning files and I wanted all provisioning file references to be consistent.

acrowthe commented:
Moved to draft:
Need to test what impact changing the install run modes has on existing sling installations

acrowthe committed:
LFS-856: Simplify CARDS run modes
Revert cards4care runmode back to cardiac_rehab for compatibility
Rename cards4kids runmode to kids

sashaandjic commented:
As per our agreement at standup today we will be supporting the following sun modes:
- **lfs** _for cards4lfs; lfs data core_
- **kids** _for cards4kids - previously called cardiac_
- **cardiac_rehab**  _will stay as is because we already have it in production_

sashaandjic commented:
re-tested

sdumitriu reviewed (changes requested):
I think we should also rename the main directory (`/cardiac-resources`).

acrowthe committed:
LFS-856: Simplify CARDS runmodes
- Rename `cardiac-resources` folder to `kids-resources`
- Adjust Cards4Kids artifact name and description to match kids labeling

acrowthe commented:
`cardiac-resources` directory renamed to `kids-resources`

sashaandjic reviewed (approved):
re-tested

sdumitriu commented on distribution/src/main/provisioning/61-cardiac-resources.txt:
```diff
@@ -18,10 +18,10 @@
 #
```
This file should be renamed as well.

sdumitriu committed:
LFS-856: Simplify CARDS runmodes
Rename the kids provisioning file
